cmd/util/flags/configflags/register.go

```diff
@@ -43,14 +44,22 @@ type Definition struct {
 
 // BindFlags binds flags from a command to Viper using the provided definitions.
 // This method should be called in command `PreRun`
-func BindFlags(cmd *cobra.Command, register map[string][]Definition) error {
+func BindFlags(cmd *cobra.Command, c config.Context, register map[string][]Definition) error {
+	seen := make(map[string]Definition)
 	for _, defs := range register {
 		for _, def := range defs {
+			// sanity check to ensure we are not binding a config key on more than one flag.
+			if dup, ok := seen[def.ConfigPath]; ok {
+				return fmt.Errorf("DEVELOPER ERROR: duplicate regsistration of config key %s for flag %s"+
+					" previously registered on on flag %s", def.ConfigPath, def.FlagName, dup.FlagName)
+			}
+			seen[def.ConfigPath] = def
```

Comment:
added to prevent bugs like the one in [cmd/util/flags/configflags/timeouts.go](https://github.com/bacalhau-project/bacalhau/pull/3959/files#diff-ddb4476b86c89712ae5ba6c7f256d7b0afcab61918271f36f3fc46e00c2bd25c) from going unnoticed.